drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -1130,16 +1136,25 @@ local heating_setpoint_limit_handler_factory = function(minOrMax)
     device:set_field(minOrMax, val)
     local min = device:get_field(setpoint_limit_device_field.MIN_HEAT)
     local max = device:get_field(setpoint_limit_device_field.MAX_HEAT)
-    if min ~= nil and max ~= nil then
-      if min < max then
-        -- Only emit the capability for RPC version >= 5 (unit conversion for
-        -- heating setpoint range capability is only supported for RPC >= 5)
-        if version.rpc >= 5 then
-          device:emit_event_for_endpoint(ib.endpoint_id, capabilities.thermostatHeatingSetpoint.heatingSetpointRange({ value = { minimum = min, maximum = max }, unit = "C" }))
-        end
-      else
-        device.log.warn_with({hub_logs = true}, string.format("Device reported a min heating setpoint %d that is not lower than the reported max %d", min, max))
+    if min == nil or max == nil then
+      return
+    end
+    if min < DEFAULT_MIN_SETPOINT then
+      min = DEFAULT_MIN_SETPOINT
+    end
+    if max > DEFAULT_MAX_SETPOINT then
+      max= DEFAULT_MAX_SETPOINT
+    end
+    if min < max then
+      -- Only emit the capability for RPC version >= 5 (unit conversion for
+      -- heating setpoint range capability is only supported for RPC >= 5)
+      if version.rpc >= 5 then
+        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.thermostatHeatingSetpoint.heatingSetpointRange({ value = { minimum = min, maximum = max }, unit = "C" }))
+        device:set_field(setpoint_limit_device_field.MIN_HEAT, min)
```

Comment:
Should these `set_field` statements be placed outside of the if statement? Even if we don't set the capability we could still set the min and max values to their limited values.